packages/test-utils/src/matches.js

```diff
@@ -6,8 +6,12 @@ import {
 import { isConstructor } from 'shared/validators'
 
 export function vmMatchesName(vm, name) {
+  const normalize = (name = '') => name.replace(/-/gi, '').toLowerCase()
```

Comment:
Wondering if this should be a globally-available utility function..?

Reply:
Wont this match `TheNavigation` and `the-navigation` with `thenavigation` ?

Reply:
Hm, I can't see anywhere else we do something similar to this. Does anywhere come to mind?
Also just noticed this vmMatchesName does not need to be exported, going to remove that.

Reply:
@dobromir-hristov It will. Will this be a problem in practice? I guess the edge case is someone has two components, one is `TheNavigation` and one is `The-Navigation`. What do you think? I don't think that's too likely.

Reply:
`TheNavigation` and `The-Navigation` I think are the same for Vue, `thenavigation` is not I think.... We can give it a test.

But you are right, its kind of a niche case.

Reply:
Ok I agree - we should match whatever Vue does.

Reply:
@dobromir-hristov all the same in SFC: https://codesandbox.io/s/festive-star-dbr9n. I think we need to remove the `h` anyway if we want to allow `FooBar` to match against `foo-bar` (which we should, since from Vue's point of view either or is fine)

Reply:
`FooBar` is not `foobar`, according to your example. But am not sure if its a problem rly..

Reply:
We can also see how `vue-devtools` does its filtering?

Reply:
For reference, this is how Vue does name resolving: https://github.com/vuejs/vue/blob/b07087deae060244fe5d9651b38dc5f1e79667a9/src/core/util/options.js#L454-L460 .

Reply:
For reference, this is how Vue does component name resolving: https://github.com/vuejs/vue/blob/b07087deae060244fe5d9651b38dc5f1e79667a9/src/core/util/options.js#L454-L460 .

Reply:
Hm. In that case I guess `FooBar` should not match `foobar`. Although not technically correct, for code simplicity I personally don't mind - the goal is to let people easily `find` a component and write tests, and I think the edge case is unlikely enough to occur.

On the flip side, I see an argument for matching Vue as closely as possible, where `FooBar` is not `foobar`. Although in the SFC example linked above, it does not seem to make a difference.

Happy to go either decision. Anyone want to chime in? For me, DX and ease of use are the goal.

Reply:
> Although in the SFC example linked above, it does not seem to make a difference.

I think it did? `<thenavigation>` didn't match either `TheNavigation` or `The-Navigation` components.

Reply:
Oh, my bad. So to sum up, `find({ name: 'fooBar' })` should match:

- `name: 'FooBar'`
- `name: 'foobar`
- `name: 'fooBar'`

Basically we don't care about the case. But the `-`. is relevant. So, don't match `name: foo-bar`. I will make the change this evening! That should solve the original issue the OP pointed out, while keeping to way Vue matches names.

Reply:
I'd say don't normalize at all and just check the three variations Vue checks. I've just tested, and `<fooBar>` only matches `FooBar` and `fooBar`, but not `foobar`.

Reply:
Yep, agreed, thanks!

Reply:
So we'll wait for this change and then re-review? @lmiller1990

Reply:
yes I will ping you guys when I update it! Within 24h.

Reply:
ok, should be good now @JessicaSachs @dobromir-hristov @afontcu